common/src/PCLPointCloud2.cpp

```diff
@@ -58,7 +58,7 @@ pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPoi
   const auto size1 = cloud1.width * cloud1.height;
   const auto size2 = cloud2.width * cloud2.height;
   //if one input cloud has no points, but the other input does, just select the cloud with points
-  switch ((bool (size1) << 1) + bool (size2))
+  switch ((static_cast<bool>(size1) << 1) + static_cast<bool>(size2))
```

Comment:
This switch seems unnecessarily complex. Can you rewrite it with if statements?

Reply:
Sorry, I may have been unclear here, I meant to also remove the bool casts and shift. The logic seems to be: If at least one cloud has no points, do the header.stamp thing and return true. Before that, if cloud2 has points but cloud1 doesn't, copy.